File: docs/guides/examples/cpp/src/data_querying_filter.cc
#include <reduct/client.h>
#include <iostream>
#include <cassert>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;

using std::chrono_literals::operator ""s;

int main() {
    // Create a client instance, then get or create a bucket
    auto client = IClient::Build("http://127.0.0.1:8383", {.api_token="my-token"});
    auto [bucket, create_err] = client->GetOrCreateBucket("datasets");
    assert(create_err == Error::kOk);

    // Query 10 photos from "imdb" entry which taken in 2006 but don't contain "Rowan Atkinson"
    auto err = bucket->Query("imdb", std::nullopt, std::nullopt, {
            .include = {{"photo_taken", "2006"}},
            .exclude = {{"name", "b'Rowan Atkinson'"}},
            .limit = 10,
    }, [](auto rec) {
        std::cout << "Name: " << rec.labels["name"] << std::endl;
        std::cout << "Photo Taken: " << rec.labels["photo_taken"] << std::endl;
        std::cout << "Gender: " << rec.labels["gender"] << std::endl;

        auto [_, read_err] = rec.ReadAll();
        assert(read_err == Error::kOk);
        return true;    // if false, the query will stop
    });

    assert(err == Error::kOk);
    return 0;
}

File: versioned_docs/version-1.9.x/examples/js/src/data_querying_filter.mjs
import { Client } from "reduct-js";
import assert from "node:assert";

// Create a client instance, then get or create a bucket
const client = new Client("http://127.0.0.1:8383", { apiToken: "my-token" });
const bucket = await client.getBucket("example-bucket");

// Query 10 photos from "imdb" entry which taken in 2006 but don't contain "Rowan Atkinson"
for await (const record of bucket.query("imdb", undefined, undefined, {
  limit: 10,
  include: { photo_taken: "2006" },
  exclude: { name: "b'Rowan Atkinson'" },
})) {
  console.log("Name", record.labels.name);
  console.log("Phot taken", record.labels.photo_taken);
  console.log("Gender", record.labels.gender);
  await record.readAsString();
}
